cmd/minikube/cmd/start.go

```diff
@@ -536,6 +536,11 @@ func validateConfig() {
 			out.V{"memory": memorySizeMB, "default_memorysize": pkgutil.CalculateSizeInMB(constants.DefaultMemorySize)})
 	}
 
+	cpuCount = viper.GetInt(cpus)
+	if cpuCount < constants.MinimumCPUS {
+		exit.UsageT("Requested cpu count {{.requested_cpus}} is less than the minimum allowed of {{.minimum_cpus}}", out.V{"requested_cpus": cpuCount, "minimum_cpus": constants.MinimumCPUS})
```

Comment:
This check looks great, but to fully resolve the issue, there is an additional complication: The `none` driver doesn't respect the value of `--cpus`, as it is not possible to programatically add a CPU to a physical machine.

This check will need an extra step, something like:

```go
import "github.com/shirou/gopsutil/cpu"

if viper.GetString(vmDriver) == constants.DriverNone {
  ci, err := cpu.Info()
  if err == nil {
    glog.Warningf("Unable to get CPU info: %v", err)
  } else {
    cpuCount = ci.Cores
  }
```

For what it's worth, the memory check suffers the same issue, but I'm not sure what the appropriate API is to solve it, so I won't ask you to do so =)

Reply:
This should probably tie in with the `Bootstrapper` somehow ?

Since the requirement comes from `kubeadm` rather than k8s

Reply:
@tstromberg  Hows this? 1217054

@afbjorklund sorry I'm not quite sure what you want me to change.

Reply:
afbjorklund was remarking that our minimum memory and CPU core requirements may change in the future depending on the bootstrapper used. For instance, k3s won't require two cores.

I'm OK with deferring per-bootstrapper requirements until we have two with different requirements.